io-engine/src/bdev/nexus/nexus_persistence.rs

```diff
@@ -119,6 +121,21 @@ impl<'n> Nexus<'n> {
                     None => nexus_info.children.push(child_info),
                 }
             }
+            PersistOp::RemoveChild {
+                child_uri,
+            } => {
+                let uuid = NexusChild::uuid(&child_uri)
+                    .expect("Failed to get child UUID.");
+
+                let idx =
+                    nexus_info.children.iter().position(|r| r.uuid == uuid);
+                if idx.is_some() {
+                    let c = nexus_info.children.remove(idx.unwrap());
+                    // Making sure that we didn't remove a healthy child.
+                    // This better be an assert and not debug_assert.
+                    assert!(!c.healthy);
```

Comment:
I don't think this is correct, we can remove healthy children

Reply:
Under what scenarios can we remove healthy children?

Reply:
There's nothing to enforce healthy children can't be removed. For example: reducing the number of volume replicas.